### PR TITLE
Remove the need for /opt/pulumi

### PR DESCRIPTION
**Pull request by ellismg**
This is an artifact of the past, where we did not ship packages via
NPM but instead linked everything. At that time, we wanted to have our
own install location for Pulumi, because it reflected how we shipped
the product.

This was pretty confusing, because we had both `build` and `install`
targets, and we'd put some binaries both in `GOBIN` and
`/opt/pulumi/bin` leading to general confusion.

Now we simplify things. We install everything into GOBIN (both our go
binaries and the language executor scripts) and when we `yarn link` to
register packages we just link the version we built into the bin
folder.

Overall, this should simplify our story and make things a little more
regular.

We no longer require any directories to be created before building
Pulumi.

### CODE REVIEW

**Comment by justinvp** [comment]
Want to update the following (which refer to `/opt/pulumi`)?
 - https://github.com/pulumi/pulumi/blob/master/CONTRIBUTING.md
 - https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/README.md

File: sdk/nodejs/Makefile [hunk 1]
```diff
@@ -29,12 +31,12 @@ build::
 	cp -R proto/. bin/proto/
 	mkdir -p bin/tests/runtime/langhost/cases/
 	find tests/runtime/langhost/cases/* -type d -exec cp -R {} bin/tests/runtime/langhost/cases/ \;
+	cd bin/ && \
+	cp ../yarn.lock . && \
+	(yarn unlink > /dev/null 2>&1 || true) && \
+	yarn link && \
+	yarn install --offline --production
```
Comment (mmdriley) [inline sdk/nodejs/Makefile]: may be easier to read this if line continuations are indented

File: sdk/python/Makefile [hunk 2]
```diff
@@ -29,24 +29,20 @@ ensure::
 	$(PIP) install pipenv==$(PIPENV_VERSION) --user
 	pipenv $(PIPENV_PYTHON_VERSION) install --dev
 	mkdir -p $(PYENVSRC)
-	
+
 build::
 	rm -rf $(PYENVSRC) && cp -R ./lib/. $(PYENVSRC)/
```
Comment (mmdriley) [inline sdk/python/Makefile]: out of scope, but `rm -rf <unquoted expansion>` is a bit scary

File: sdk/nodejs/Makefile (hunk 1 above)
Comment (ellismg) [reply sdk/nodejs/Makefile]: Sure!

File: sdk/python/Makefile (hunk 2 above)
Comment (ellismg) [reply sdk/python/Makefile]: Fixed it since I was in the area.

**Comment by joeduffy** [comment]
@ellismg Almost ready to land this one?

**Comment by joeduffy** [comment]
@ellismg Ping on this one?

**Comment by ellismg** [comment]
Sorry for missing this.  The upstack repos will need similar changes.  Let me prepare them and then push the entire thing through tomorrow.

**Pull request closed without merging**